workbench-core/authorization/src/dynamicAuthorization/ddbDynamicAuthorizationPermissionsPlugin.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AwsService, JSONValue } from '@aws/workbench-core-base';
+import { Action } from '../action';
+import { IdentityPermissionCreationError } from '../errors/identityPermissionCreationError';
+import { ThroughputExceededError } from '../errors/throughputExceededError';
+import { Effect } from '../permission';
+import {
+  CreateIdentityPermissionsRequest,
+  CreateIdentityPermissionsResponse
+} from './dynamicAuthorizationInputs/createIdentityPermissions';
+import {
+  DeleteIdentityPermissionsRequest,
+  DeleteIdentityPermissionsResponse
+} from './dynamicAuthorizationInputs/deleteIdentityPermissions';
+import {
+  GetDynamicOperationsByRouteRequest,
+  GetDynamicOperationsByRouteResponse
+} from './dynamicAuthorizationInputs/getDynamicOperationsByRoute';
+import {
+  GetIdentityPermissionsByIdentityRequest,
+  GetIdentityPermissionsByIdentityResponse
+} from './dynamicAuthorizationInputs/getIdentityPermissionsByIdentity';
+import {
+  GetIdentityPermissionsBySubjectRequest,
+  GetIdentityPermissionsBySubjectResponse
+} from './dynamicAuthorizationInputs/getIdentityPermissionsBySubject';
+import { IdentityPermission, IdentityType } from './dynamicAuthorizationInputs/identityPermission';
+import { IsRouteIgnoredRequest, IsRouteIgnoredResponse } from './dynamicAuthorizationInputs/isRouteIgnored';
+import {
+  IsRouteProtectedRequest,
+  IsRouteProtectedResponse
+} from './dynamicAuthorizationInputs/isRouteProtected';
+import { DynamicAuthorizationPermissionsPlugin } from './dynamicAuthorizationPermissionsPlugin';
+
+export class DDBDynamicAuthorizationPermissionsPlugin implements DynamicAuthorizationPermissionsPlugin {
+  private _awsService: AwsService;
+
+  public constructor(config: { awsService: AwsService }) {
```

Comment:
Can we declare the required dependencies instead of throwing the entire `AwsService` object in here? It looks like we're only using the database service.